packages/common/core-interfaces/src/error.ts

```diff
@@ -34,6 +34,16 @@ export const FluidErrorTypes = {
 	 * Error indicating an API is being used improperly resulting in an invalid operation.
 	 */
 	usageError: "usageError",
+
+	/**
+	 * Error indicating that two Fluid layers are incompatible. For instance, if the Loader layer is
+	 * not compatible with the Runtime layer, the container create / load will fail with an error of this type.
+	 * In most cases, the layer compatibility validation happens during container load / create causing it to
+	 * fail with this error type.
+	 * In some cases such as for the Runtime and DataStore layer compatibility, the incompatibility may be detected
+	 * during data store loads. In such cases, the data store load will fail with this error type.
```

Comment:
Nit (summary should be a brief semantic description, and remaining notes should go in other blocks)
```suggestion
	 * Error indicating that two Fluid layers are incompatible.
	 * @remarks For instance, if the Loader layer is
	 * not compatible with the Runtime layer, the container create / load will fail with an error of this type.
	 * In most cases, the layer compatibility validation happens during container load / create causing it to
	 * fail with this error type.
	 * In some cases such as for the Runtime and DataStore layer compatibility, the incompatibility may be detected
	 * during data store loads. In such cases, the data store load will fail with this error type.
```